app/models/katello/host/subscription_facet.rb

```diff
@@ -297,6 +297,18 @@ def backend_update_needed?
         false
       end
 
+      def self.populate_fields_from_facts(host, parser, _type, _source_proxy)
+        has_convert2rhel = parser.facts.key?('conversions.env.CONVERT2RHEL_THROUGH_FOREMAN')
+        # Add in custom convert2rhel fact if system was converted using convert2rhel through Katello
+        # We want the value nil unless the custom fact is present otherwise we get a 0 in the database which if debugging
+        # might make you think it was converted2rhel but not with satellite, that is why I have the tenary below.
+        facet = host.subscription_facet || host.build_subscription_facet
```

Comment:
As discussed on Matrix: building a subscription facet is wrong because we have unregistered hosts (like the Katello server itself) that should remain unregistered. Previously we didn't update the facts, but @evgeni just introduced code to do cause. This causes https://community.theforeman.org/t/katello-nightly-rpm-pipeline-2097-failed/39127 a regression.